source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -64,13 +89,39 @@ bool LocalRateLimiterImpl::requestAllowed() const {
     synchronizer_.syncPoint("allowed_pre_cas");
 
     // Loop while the weak CAS fails trying to subtract 1 from expected.
-  } while (!tokens_.compare_exchange_weak(expected_tokens, expected_tokens - 1,
-                                          std::memory_order_relaxed));
+  } while (!tokens.tokens.compare_exchange_weak(expected_tokens, expected_tokens - 1,
+                                                std::memory_order_relaxed));
 
   // We successfully decremented the counter by 1.
   return true;
 }
 
+bool LocalRateLimiterImpl::requestAllowed() const { return requestAllowedHelper(tokens_); }
+
+void LocalRateLimiterImpl::fillDescriptor(
```

Comment:
Can you structure this so it's done in the constructor? This is only done once and the code will be easier to understand if it's done there.

Reply:
That imply parsing envoy::extensions::common::ratelimit::v3::LocalRateLimitDescriptor in here and also maintaining the vector<LocalRateLimitDescriptor> here. We could have gotten away with the map we already have but as we need ordering when matching the LocalRateLimitDescriptor, we need to maintain this vector.. what do you think?